### PR TITLE
Add error molecule pattern

### PR DESCRIPTION
**Pull request by discodavey**

### CODE REVIEW

File: src/patterns/molecules/error/error.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.error {
+  text-align: center;
+}
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.scss]: New line.

File: src/patterns/molecules/error/error.stories.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+      },
+    details:
+      {
+        text: 'Something has gone seriously wrong. We’ve logged the problem and will look into it.',
+      },
+  },
+);
+
+export const ErrorAr = () => error(
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.stories.js:46]: Missing `dir` and `lang` attributes.

File: src/patterns/molecules/error/error.stories.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      },
+    details:
+      {
+        text: ' لقد حدث خطأ ما بشكل خطير. لقد سجلنا المشكلة وسوف ننظر فيها.',
+      },
+  },
+);
+
+export const ErrorJa = () => error(
+  {
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.stories.js:71]: Missing `lang` attribute.

**Review by davidcmoulton — COMMENTED** [review]
Not many styles in this. Is that intentional because they're for spacing that's not ported over yet, or some other reason?

**Comment by discodavey** [comment]
I have left out the mixins that need the logical properties mixin but have put in a TODO for each one.

File: .storybook/config.js [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import { addDecorator, addParameters, configure } from '@storybook/html';
 import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';
-import centered from '@storybook/addon-centered/html';
```
Comment (discodavey) [inline .storybook/config.js:3]: Removed as confusion to patterns that centered themselves.

File: src/patterns/molecules/error/error.scss [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+.error__image {
+  @include space.block($end: baseline.for-units(4));
+
+  @supports (display: flex) {
+    display: block;
+  }
+}
+
+.error__heading {
+  @include space.block($end: rem.calculate(baseline.for-units(2)));
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.scss:42]: (Out of scope.) Wondering if we should have a mixin for `rem.calculate(baseline.for-units(2))` as it seems a reasonably common use case.

File: src/patterns/molecules/error/error.scss [hunk 6]
```diff
@@ -0,0 +1,45 @@
+@use "../../../shared-styles/normalize"; // Always used first
+@use "../../../shared-styles/baseline";
+@use "../../../shared-styles/grid";
+@use "../../../shared-styles/media-query";
+@use "../../../shared-styles/rem";
+@use "../../../shared-styles/scale";
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.scss]: `scale ` is not used so should be removed.

File: src/shared-styles/_size.scss [hunk 7]
```diff
@@ -3,6 +3,16 @@
...
+@mixin max-inline-size($size) {
+  @include validation.expect_single_value($size) {
+    @include _constrain-inline-size($size, max);
+  }
+}
+
+@mixin _constrain-inline-size($size, $extreme) {
+  // TODO: Add Logical Properties when tests have been put in place.
+}
+
```
Comment (davidcmoulton) [inline src/shared-styles/_size.scss]: Can be deleted, as now handled by mixins at https://github.com/libero/storybook/blob/master/src/shared-styles/_size.scss#L26 and https://github.com/libero/storybook/blob/master/src/shared-styles/_size.scss#L34.

File: src/shared-styles/_spacing.scss [hunk 8]
```diff
@@ -0,0 +1,66 @@
+@use "logical";
```
Comment (davidcmoulton) [inline src/shared-styles/_spacing.scss]: The whole file `_spacing.scss` can be removed as it's superseded by `_space.scss`.

File: src/shared-styles/_validation.scss [hunk 9]
```diff
@@ -0,0 +1,17 @@
+@mixin expect_at_most($value, $maxLength, $message: "Expected at most #{$maxLength} values") {
```
Comment (davidcmoulton) [inline src/shared-styles/_validation.scss]: The whole file `_validation.scss` can be removed as it's superseded by `_validate.scss`.

File: src/patterns/molecules/error/error.scss [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+@use "../../../shared-styles/size";
+@use "../../../shared-styles/space";
+@use "../../../shared-styles/typography";
+
+.error {
+  text-align: center;
+  @include size.max-inline-size(grid.$max-inline-size);
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.scss:12]: (Out of scope) `@include size.max-inline-size()` seems a bit clumsy.

Renaming the dimension constraint mixins could give us either:
1.
  - `@include size.min-inline()`
  - `@include size.max-inline()`

or

2.

  - `@include size.inline-min()`
  - `@include size.inline-max()`

Think I prefer 1 as it more closely matches the underlying css property names. Thoughts? (Would be a separate PR.)

File: src/patterns/molecules/error/error.scss [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  @include space.margin(auto, inline);
+  @include space.padding(grid.$column_gap, inline);
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.scss]: `grid.$column-gap` (hyphen).

File: src/patterns/molecules/error/error.scss [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+  @include space.block($start: rem.calculate(baseline.for-units(12)), $end: rem.calculate(baseline.for-units(12)));
+  @include media-query.mq($from: medium) {
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.scss:16]: (Out of scope.) While I'm on a roll with the naming revisionism... 😅  I don't much like `@include media-query.mq()` either. `@include media.query()` would read better, but not sure how feasible this is with how we're using the `mq` library. Will have a play and see if can get something sensible from it (into a separate PR).

File: src/patterns/molecules/error/error.scss [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  > :last-child {
+    @include space.margin(0, block-end);
+  }
+
+}
+
+.error__image {
+  @include space.block($end: baseline.for-units(4));
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.scss]: Can just be `@include space.block()` as [the `block()` mixin](https://github.com/libero/storybook/blob/master/src/shared-styles/_space.scss#L32) has `baseline.for-units(4)` as its default `$end` value.

**Review by davidcmoulton — CHANGES_REQUESTED** [review]
Don't think it needs to be addressed now, but noting that we also have an [`error` Sass module](https://github.com/libero/storybook/blob/master/src/shared-styles/_error.scss) in addition to this error pattern. If this causes confusion, I'd prefer to rename this pattern rather than the sass module, as the `error.throw()`formulation for the sass module works well.

File: src/patterns/molecules/error/error.stories.js [hunk 14]
```diff
@@ -0,0 +1,86 @@
+import error from './error.twig';
+import '../heading/heading.scss';
+import './error.scss';
+import image from './images/5xx.svg';
+
+export default {
+  title: 'Molecules|Error',
+};
```
Comment (davidcmoulton) [inline src/patterns/molecules/error/error.stories.js:13]: To disable the `centered` add-on just for the stories in this file, modify the default export like this:
```
export default {
  title: 'Molecules|Error',
  parameters: {
    centered: {
      disable: true,
    },
  },
};
```
(Reference: https://github.com/storybookjs/storybook/pull/7709.)

File: .storybook/config.js (hunk 4 above)
Comment (davidcmoulton) [reply .storybook/config.js:3]: I've found the way of disabling `centered` on a per-story basis, see https://github.com/libero/storybook/pull/30/files#r349164152.

For this to work, first you'll need to revert the changes in fa8b5e9 (and then afterwards, to fix up your local environment, run `make install && make build` to get the node module back into your local environment).

File: src/patterns/molecules/error/error.scss (hunk 5 above)
Comment (discodavey) [reply src/patterns/molecules/error/error.scss:42]: Possibly but I always think having mixins within mixins could confuse and how many mixins we want to nest before it's too much?

File: src/patterns/molecules/error/error.scss (hunk 10 above)
Comment (discodavey) [reply src/patterns/molecules/error/error.scss:12]: I think number one is better as it matches CSS styling like max-width or max-height

File: src/patterns/molecules/error/error.scss (hunk 5 above)
Comment (davidcmoulton) [reply src/patterns/molecules/error/error.scss:42]: My thinking is one well chosen name could make the code easier to read, rather than having to mentally parse the nested call in the style rules on each occurrence 🤔

**Review by davidcmoulton — APPROVED** [review]
